src/operators/flatten.ts

```diff
@@ -49,10 +49,11 @@ export class FlattenOperator implements Operator {
   }
 
   handleData(data: any[]): any[] | null {
-    const flattenedData = data;
-    flattenedData[this.index] = this.flattenHelper(flattenedData[this.index]);
+    const flattened = this.flattenHelper(data[this.index]);
```

Comment:
maybe note that `flattenHelper` is pure (no side effects) ?

Reply:
actually the helper function isn't "pure" but the way it's called isn't mutating the data.

Reply:
Yes, I was just writing the comments now.  The fact that the `node` param is not passed has it create a new object under the hood.  I'm going to tweak this code a bit so that's more obvious.